drivers/sensor/mcux_tempmon/mcux_tempmon.c

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (c) 2023 Centralp
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT nxp_tempmon
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(mcux_tempmon, CONFIG_SENSOR_LOG_LEVEL);
+
+/* OTP Controller Analog Register 1 for calibration values */
+#define OCOTP_ANA1_ADDR             0x401F44E0
+#define OCOTP_ANA1_ROOM_COUNT_SHIFT 20
+#define OCOTP_ANA1_ROOM_COUNT_MASK  (BIT_MASK(12) << OCOTP_ANA1_ROOM_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_COUNT_SHIFT  8
+#define OCOTP_ANA1_HOT_COUNT_MASK   (BIT_MASK(12) << OCOTP_ANA1_HOT_COUNT_SHIFT)
+#define OCOTP_ANA1_HOT_TEMP_SHIFT   0
+#define OCOTP_ANA1_HOT_TEMP_MASK    (BIT_MASK(8) << OCOTP_ANA1_HOT_TEMP_SHIFT)
+
+#define TEMPMON_ROOM_TEMP 25.0f
+
+struct mcux_tempmon_data {
+	uint8_t hot_temp;
+	uint16_t hot_cnt;
+	uint16_t room_cnt;
+};
+
+struct mcux_tempmon_config {
+	TEMPMON_Type *base;
+};
+
+static int mcux_tempmon_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct mcux_tempmon_config *cfg = dev->config;
+
+	if (chan != SENSOR_CHAN_ALL && chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	/* Start the measure */
+	cfg->base->TEMPSENSE0 |= TEMPMON_TEMPSENSE0_MEASURE_TEMP_MASK;
+
+	return 0;
+}
+
+static int mcux_tempmon_channel_get(const struct device *dev, enum sensor_channel chan,
+				    struct sensor_value *val)
+{
+	const struct mcux_tempmon_data *data = dev->data;
+	const struct mcux_tempmon_config *cfg = dev->config;
+	uint16_t temp_cnt;
+	float temp;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	if (!(cfg->base->TEMPSENSE0 & TEMPMON_TEMPSENSE0_FINISHED_MASK)) {
+		/* Measure is not ready */
+		return -EINVAL;
+	}
+
+	temp_cnt = (cfg->base->TEMPSENSE0 & TEMPMON_TEMPSENSE0_TEMP_CNT_MASK) >>
+		   TEMPMON_TEMPSENSE0_TEMP_CNT_SHIFT;
+	temp = (float)data->hot_temp - ((float)temp_cnt - (float)data->hot_cnt) *
```

Comment:
Not familiar with this peripheral so I may be off here, but I was looking into https://www.nxp.com/docs/en/application-note/AN5215.pdf and it seems that there’s an updated single-point calibration equation that yields more accurate readings. Have you considered or tried that one, if applicable?

Reply:
I don't know if the `offset` mentioned in section 3.2 is also applicable for i.MXRT (the AN is written for i.MX6 SoC) and I'm not equipped to check such measurements.

Reply:
ok, not a blocking comment from my side